lib/xero_gateway/partner_app.rb

```diff
@@ -2,13 +2,16 @@ module XeroGateway
   class PartnerApp < Gateway
 
     class CertificateRequired < StandardError; end
+    class UserAgentRequired < StandardError; end
 
     NO_PRIVATE_KEY_ERROR_MESSAGE = "You need to provide your private key (corresponds to the public key you uploaded at api.xero.com) as :private_key_file (should be .crt or .pem files)"
+    NO_USER_AGENT_ERROR_MESSAGE = "a unique User-Agent header is required for partner apps and should be supplied as :user_agent"
 
     def_delegators :client, :session_handle, :renew_access_token, :authorization_expires_at
 
     def initialize(consumer_key, consumer_secret, options = {})
       raise CertificateRequired.new(NO_PRIVATE_KEY_ERROR_MESSAGE) unless options[:private_key_file]
+      raise UserAgentRequired.new(NO_USER_AGENT_ERROR_MESSAGE) unless options[:user_agent]
```

Comment:
This is going to break existing installs. Instead, can we output a warning?

Reply:
good point - we can indeed.  I believe this is a requirement for new partner apps, so I'm not sure if you'd prefer to leave this in, but only for a new major version of this gem.  For now, I'll back it off to output a warning.